src/Workspaces/Core/Portable/SolutionCrawler/IIncrementalAnalyzerExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.SolutionCrawler
+{
+    internal static partial class IIncrementalAnalyzerExtensions
+    {
+        public static BackgroundAnalysisScope GetOverriddenBackgroundAnalysisScope(this IIncrementalAnalyzer incrementalAnalyzer, OptionSet options, BackgroundAnalysisScope defaultBackgroundAnalysisScope)
+        {
+            // Unit testing analyzer has special semantics for analysis scope.
+            if (incrementalAnalyzer is UnitTestingIncrementalAnalyzer unitTestingAnalyzer)
```

Comment:
@mavasani As I'd mentioned offline, I think it would be better if this were controlled in a way that does not require special casing the unit testing analyzer.

One thought is that perhaps a setting could be added via [`IncrementalAnalyzerProviderMetadata`](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=lodpzbnjgfvf&file=SolutionCrawler%5CIncrementalAnalyzerProviderMetadata.cs) (similar to [`HighPriorityForActiveFile`](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=fgwspladb5yi&file=SolutionCrawler%5CIncrementalAnalyzerProviderMetadata.cs&rightSymbol=lodpzbnjgfvf)). The metadata can also be provided via MEF if required ([`ExportIncrementalAnalyzerProviderAttribute`](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=bmkg84v0jaki&file=SolutionCrawler%5CExportIncrementalAnalyzerProviderAttribute.cs&rightSymbol=cxjbc63yhgfx)). However, the unit testing analyzer is not exposed via MEF - so if you wanted to limit this to just the unit testing analyzer, you could make the opt out setting only available to analyzers that register via `ISolutionCrawlerRegistrationService` and supply `IncrementalAnalyzerProviderMetadata`...

Not urgent - but hopefully we can look into this at some point in the future...

Reply:
Thanks @shyamnamboodiripad. I will try to get to this in a follow-up PR.